sound/soc/sof/loader.c

```diff
@@ -61,11 +60,13 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 		/* process structure data */
 		switch (ext_hdr->type) {
 		case SOF_IPC_EXT_DMA_BUFFER:
+			ret = 0;
 			break;
 		case SOF_IPC_EXT_WINDOW:
 			ret = get_ext_windows(sdev, ext_hdr);
 			break;
 		default:
+			ret = 0;
```

Comment:
please don't submit two PRs which touch the same lines of code. It will lead to conflicts.
Also why is the point of changing the return value? get _ext_windows can only return 0, -EINVAL and -ENOMEM and the last two are trapped. doing those additional inits does not bring any added value. If you want to be explicit and not force the reader to reconstruct the flow, move this outside of each switch case so that the value is reset for each new header.

Reply:
> please don't submit two PRs which touch the same lines of code

I know, but I do as you wish in PR https://github.com/thesofproject/linux/pull/1582

> move this outside of each switch case

> can we reset ret to 0 outside the switch?

Do you really like such a solution? I prefer my way or make something like
```
switch(...) {
  case 1:
  case 2:
    [..]
}
```

Because it is the most explicit solution, it binds together case=>action in straightforward way, so reader can't be confused.